### ISSUE TITLE
<concepts> header causes build breaks in openage

### ISSUE THREAD

**Issue by duanqn** [SFTtech/openage#1401]
From https://github.com/duanqn/openage/runs/3383699397?check_suite_focus=true

```
In file included from /mnt/openage/.bin/g++-release-Oauto-sanitize-none/nyan-external/source/nyan/type.h:15,
                 from /mnt/openage/.bin/g++-release-Oauto-sanitize-none/nyan-external/source/nyan/ast.h:18,
                 from /mnt/openage/.bin/g++-release-Oauto-sanitize-none/nyan-external/source/nyan/ast.cpp:3:
/mnt/openage/.bin/g++-release-Oauto-sanitize-none/nyan-external/source/nyan/util/flags.h:5:10: fatal error: concepts: No such file or directory
    5 | #include <concepts>
      |          ^~~~~~~~~~
compilation terminated.
make[3]: *** [nyan-external/bin/nyan/CMakeFiles/nyan.dir/build.make:83: nyan-external/bin/nyan/CMakeFiles/nyan.dir/ast.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:3501: nyan-external/bin/nyan/CMakeFiles/nyan.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
```

**Comment by duanqn** [comment]
Sounds like a huge task to move to C++20... But I've succeeded with Ubuntu.
https://github.com/duanqn/openage/runs/3389004021?check_suite_focus=true

**Comment by TheJJ** [comment]
Yes, nyan now requires c++20, so openage also has to support this :)

**Comment by TheJJ** [comment]
Ah the issue is now in the openage repo, didn't know issue transfers are even posible. We'll leave it open until we've fixed it.

**Comment by TheJJ** [comment]
Ok, now fixed with #1402 :)
